src/daft-functions/src/numeric/sign.rs

```diff
@@ -11,29 +11,38 @@ use serde::{Deserialize, Serialize};
 
 use super::{evaluate_single_numeric, to_field_single_numeric};
 
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
-pub struct Sign {}
+macro_rules! sign {
+    ($name:ident, $variant:ident) => {
+        #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+        pub struct $variant;
 
-#[typetag::serde]
-impl ScalarUDF for Sign {
-    fn as_any(&self) -> &dyn std::any::Any {
-        self
-    }
+        #[typetag::serde]
+        impl ScalarUDF for $variant {
+            fn as_any(&self) -> &dyn std::any::Any {
+                self
+            }
+            fn name(&self) -> &'static str {
+                stringify!($name)
+            }
 
-    fn name(&self) -> &'static str {
-        "sign"
-    }
+            fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+                to_field_single_numeric(self, inputs, schema)
+            }
 
-    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
-        to_field_single_numeric(self, inputs, schema)
-    }
+            fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
+                evaluate_single_numeric(inputs, Series::$name)
+            }
+        }
 
-    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
-        evaluate_single_numeric(inputs, Series::sign)
-    }
+        #[must_use]
+        pub fn $name(input: ExprRef) -> ExprRef {
+            ScalarFunction::new($variant, vec![input]).into()
+        }
+    };
 }
 
-#[must_use]
-pub fn sign(input: ExprRef) -> ExprRef {
-    ScalarFunction::new(Sign {}, vec![input]).into()
-}
+sign!(sign, Sign);
+sign!(signum, Signum);
+sign!(negate, Negate);
+sign!(negative, Negative);
+sign!(positive, Positive);
```

Comment:
instead of adding multiple impls here, I'd prefer to have a single implementation, then have the frontends (dataframe, spark, sql) handle multiple aliases.